hera_sim/foregrounds.py

```diff
@@ -7,61 +7,48 @@
 
 import aipy
 import numpy as np
-from scipy.interpolate import RectBivariateSpline
 
 from . import noise
 from . import utils
 
 
-def diffuse_foreground(Tsky, lsts, fqs, bl_len_ns, bm_poly=noise.HERA_BEAM_POLY, scalar=30.,
-                       fr_width=None, fr_max_mult=2.0):
-
+def diffuse_foreground(Tsky_mdl, lsts, fqs, bl_vec, bm_poly=noise.HERA_BEAM_POLY,
```

Comment:
If you set Tsky_mdl not as the first parameter, it will make it possible to call it from the simulator class with Tsky_mdl as a kwarg, rather than requiring it as another parameter.

Reply:
hmm, not sure how that would work, b/c `kwargs` is an unordered dictionary... if you can paste a code snippet below of what you mean it might help enlighten me

Reply:
If the signature of ``diffuse_foreground`` is ~

```python
def diffuse_foreground(lsts, fqs, bl_vec, Tsky_mdl=None, bm_poly=noise.HERA_BEAM_POLY,...):
    if Tsky_mdl is None:
        raise TypeError("Tsky_mdl is a required parameter of diffuse_foreground")
    ...
```

then in the ``Simulator`` class, the ``add_foreground`` function can leave off the ``Tsky_mdl`` argument explicitly, i.e.

```python
@_model()
    def add_foregrounds(self, model, **kwargs):
        for ant1, ant2, pol, blt_ind, pol_ind in self._iterate_antpair_pols():
            lsts = self.data.lst_array[blt_ind]
            bl_len_m = np.abs(self.antpos[ant1][0] - self.antpos[ant2][0])  # E-W baseline length

            self.data.data_array[blt_ind, 0, :, pol_ind] += model(
                lsts=lsts,
                fqs=self.data.freq_array[0] * 1e-9,
                bl_len_ns=bl_len_m * 1e9 / 3e8,
                **kwargs
            )
```
I'm not _absolutely_ convinced this is better than the way you've done it. It just seems more consistent with how the rest of the ``add_`` functions are implemented.

Reply:
Sure thing